source/kubernetes/manage-app/control-healthchecks/index.html.md

```diff
@@ -0,0 +1,200 @@
+---
+title: Control App healthchecks and kubernetes probes
+weight: 49
+layout: multipage_layout
+---
+
+# Healthchecks
+
+Kubernetes will run various probes against your app to decide:
+
+1. Whether the app has started
+2. Whether the app should be served traffic
+3. Whether your app should be restarted (by terminating the pod, and starting
+   it again)
+
+By default all of these checks are run against the `/healthcheck/live`
+endpoint.
+
+You can control which endpoints kubernetes will hit for each check.
+
+See the section [Kubernetes Probes](#kubernetes-probes) later on this page for
+a description of how the probes work, and how they will affect your app.
+
+## Choosing a different healthcheck endpoint
+
+If you wish to use a different endpoint for any of the types of [Kubernetes
+Probes](#kubernetes-probes) you can do so as follows:
+
+In [the app-config Helm chart in
+govuk-helm-charts](https://github.com/alphagov/govuk-helm-charts/tree/main/charts/app-config),
+in `values-<environment>.yaml` you can add the configuration key
+`kubernetesProbeEndpoints` into the helmValues of your applications
+configuration:
+
+```
+kubernetesProbeEndpoints:
+    startupProbe: "/healthcheck/ready"
+    livenessProbe: "/healthcheck/live"
+    readinessProbe: "/healthcheck/foo"
+```
+
+You do not need to add all 3 and could override just one. For example in [pull
+request #3406](https://github.com/alphagov/govuk-helm-charts/pull/3406/) the
+configuration for static, and draft-static, was updated in integrtion so that
+the configuration of them looked as follows and used `/healthcheck/ready` as a
+startupProbe only (leaving the liveness and readiness probes as the default
+`/healthcheck/live`:
+
+```
+  - name: static
+    helmValues:
+      arch: arm64
+      appResources:
+        ...SNIP...
+      ingress:
+        ...SNIP...
+      kubernetesProbeEndpoints:
+        startupProbe: "/healthcheck/ready"
+```
+
+## Kubernernetes Probes
+
+When kubernetes executes a Pod, it will perform 3 types of [Kubernetes
+Probes](https://kubernetes.io/docs/concepts/configuration/liveness-readiness-startup-probes/#startup-probe).
+
+### Startup Probe
+
+As soon as the pod launches kubernetes will start querying the startup probe.
```

Comment:
Unless there is an `initialDelaySeconds` defined later down the line in the helm chart.